setuptools/tests/test_config.py

```diff
@@ -703,6 +703,22 @@ def test_entry_points(self, tmpdir):
         with get_dist(tmpdir) as dist:
             assert dist.entry_points == expected
 
+    def test_data_files(self, tmpdir):
+        fake_env(
+            tmpdir,
+            '[options.data_files]\n'
+            'cfg =\n'
+            '      a/b.conf\n'
+            '      c/d.conf\n'
+            'data = e/f.dat, g/h.dat\n'
+        )
+
+        with get_dist(tmpdir) as dist:
+            assert dist.data_files == [
```

Comment:
I believe this is failing because `dist.data_files` gets parsed first to a dict (with arbitrary order on Py < 3.6), then turned into this key/value list thing.

I would look at what the other tests do, but I think a quick fix would be this:

```python
expected = {
    'cfg': ['a/b.conf', 'c/d.conf'],
    'data': ['e/f.dat', 'g/h.dat']
}

with get_dist(tmpdir) as dist:
    # Order independent comparison
    assert dict(*dist.data_files) == expected
```

Reply:
I committed the fix based on your suggestion but data type of expected result is kept.